### PR TITLE
Add RegistryEntryAddedCallback.allEntries

### PR DESCRIPTION
**Pull request by modmuss50**
RegistryEntryAddedCallback.allEntries invokes a callback for all previous and future entries in a registry. This is a common bit of boiler plate that many mods use.

### CODE REVIEW

File: fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/RegistryEntryAddedCallback.java [hunk 1]
```diff
@@ -16,17 +16,52 @@
...
+	 *
+	 * @param registry the registry to get the event for
+	 * @return the event
+	 */
 	static <T> Event<RegistryEntryAddedCallback<T>> event(Registry<T> registry) {
 		return ListenableRegistry.get(registry).fabric_getAddObjectEvent();
 	}
+
+	/**
+	 * Register a callback for all present and future entries in the registry.
```
Comment (Player3324) [inline fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/RegistryEntryAddedCallback.java:55]: I think this should have a reentry/recursion warning - if a mod adds from within the consumer it'll see an invocation for itself

Comment (modmuss50) [reply fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/RegistryEntryAddedCallback.java:55]: Yes, this was by design and I think correct/expected behaviour. Maybe just a note in the javadocs to point this out would be ok?

Comment (Player3324) [reply fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/RegistryEntryAddedCallback.java:55]: Yes that's what I meant, the note references the Javadoc section but I should have specified.

Comment (modmuss50) [reply fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/RegistryEntryAddedCallback.java:55]: Added a note about it. my bad I though you were suggesting to catch/prevent it.

**Review by Player3324 — APPROVED** [review]

**Review by apple502j — APPROVED** [review]

**Pull request merged**